extensions/cornerstone-dicom-sr/src/utils/getFilteredCornerstoneToolState.ts

```diff
@@ -0,0 +1,111 @@
+import OHIF from '@ohif/core';
+import * as cornerstone3DTools from '@cornerstonejs/tools';
+const { log } = OHIF;
+
+function getFilteredCornerstoneToolState(
+  measurementData,
+  additionalFindingTypes
+) {
+  const filteredToolState = {};
+
+  function addToFilteredToolState(annotation, toolType) {
+    if (!annotation.metadata?.referencedImageId) {
+      log.warn(
+        `[DICOMSR] No referencedImageId found for ${toolType} ${annotation.id}`
+      );
+      return;
+    }
+
+    const imageId = annotation.metadata.referencedImageId;
+
+    if (!filteredToolState[imageId]) {
+      filteredToolState[imageId] = {};
+    }
+
+    const imageIdSpecificToolState = filteredToolState[imageId];
+
+    if (!imageIdSpecificToolState[toolType]) {
+      imageIdSpecificToolState[toolType] = {
+        data: [],
+      };
+    }
+
+    const measurementDataI = measurementData.find(
+      md => md.uid === annotation.annotationUID
+    );
+    const toolData = imageIdSpecificToolState[toolType].data;
+
+    let finding;
+    const findingSites = [];
+
+    // NOTE -> Any kind of freetext value abuses the DICOM standard,
+    // As CodeValues should map 1:1 with CodeMeanings.
+    // Ideally we would actually use SNOMED codes for this.
```

Comment:
I think I wrote this, but I don't strictly think this is true. Its just that we are referencing our own definition (although it does suck its only burried in dcmjs CS3D code rather than actually published somewhere).